### PR TITLE
refactor depositContractAddress to remove the optional nature of the field

### LINKED ISSUE #3730: make Eth2NetworkConfiguration.eth1DepositContractAddress a simple Eth1Address rather than an Optional<Eth1Address>.
review comments suggested making `eth1DepositContractAddress` in `Eth2NetworkConfiguration` non optional, but it's not a small change due to where it's used, so this captures that intent.

ref. #3724

### PR DESCRIPTION
**Pull request by rolfyone**
fixes #3730

Signed-off-by: Paul Harris <paul.harris@consensys.net>
## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java [hunk 1]
```diff
@@ -38,16 +38,16 @@
...
     this.jsonProvider = jsonProvider;
     this.configProvider = configProvider;
     this.depositContractResponse = Optional.empty();
     this.depositContractAddress =
-        depositContractAddress
-            .map(Eth1Address::toHexString)
-            .orElse("0xdddddddddddddddddddddddddddddddddddddddd");
+        depositContractAddress == null
+            ? "0xdddddddddddddddddddddddddddddddddddddddd"
+            : depositContractAddress.toHexString();
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java]: Do we still need this? It's very weird to be returning `0xdddddddddddddddddddddddddddddddddddddddd` and we should now always have a deposit contract as it's a required value in the network config file.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java]: only in the integration tests from memory. I can probably just investigate the tests and see why they broke but i figured this was equivalent to what was there already... thoughts?

Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java]: Would be nice to get rid of it, but don't spend too much time, it's not a big deal.

**Pull request merged**